src/http/one/Tokenizer.cc

```diff
@@ -78,32 +62,17 @@ Http::One::Tokenizer::qdText(SBuf &returnedToken, const bool http1p0)
              */
             static const CharacterSet qPairChars = CharacterSet::HTAB + CharacterSet::SP + CharacterSet::VCHAR + CharacterSet::OBSTEXT;
             SBuf escaped;
-            if (!prefix(escaped, qPairChars, 1)) {
-                returnedToken.clear();
-                restoreLastCheckpoint();
-                return false;
-            }
-            returnedToken.append(escaped);
-            continue;
+            if (!tok.prefix(escaped, qPairChars, 1))
+                break;
 
-        } else if (skip('"')) {
-            break; // done
-
-        } else if (atEnd()) {
-            // need more data
-            returnedToken.clear();
-            restoreLastCheckpoint();
-            return false;
+            returnedToken.append(escaped);
+        } else {
+            debugs(24, 8, "invalid bytes for set " << tokenChars.name);
+            break;
         }
-
-        // else, we have an error
-        debugs(24, 8, "invalid bytes for set " << tokenChars.name);
-        returnedToken.clear();
-        restoreLastCheckpoint();
-        return false;
     }
 
-    // found the whole string
-    return true;
+    returnedToken.clear();
```

Comment:
Please avoid touching/modifying the function parameter when returning false.

Reply:
Not done: The following return statement (at least) modifies returnedToken when returning false: https://github.com/squid-cache/squid/blob/7c3462ebbc67447672095099c5965b67343e8ffc/src/http/one/Tokenizer.cc#L19

Reply:
Done at 48dcb7d.

Reply:
Still not done. For example:

https://github.com/squid-cache/squid/blob/3cdffa2b3990656edfb75142a30e7e85693192fe/src/http/one/Tokenizer.cc#L48

I know the new parseQuotedStringSuffix() function is a helper function, but I think it should follow the same "no parameter modification on false result" principle because there are no compelling reasons _not_ to follow it. The alternative is to clearly document that parseQuotedStringSuffix() is special, but I do not think introducing this general API exception is worth it.

Reply:
> Still not done

Done at 3ae2217.